dataplane/forwarding/fwdbuilder/action.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fwdbuilder contains builders for varius proto types.
+package fwdbuilder
+
+import (
+	"encoding/binary"
+
+	fwdpb "github.com/openconfig/lemming/proto/forwarding"
+)
+
+type actionDescBuilder interface {
```

Comment:
Is this a builder type or a mutator type? Its methods seems to have the capability of mutating the given ActionDesc rather than returning a built object which is what I expected when I started reading the code.

Reply:
because the protobuf oneof generates an unexported interface, it's easier for this to mutate the parent on build